docs/design/proposals/late-initialization/late-initialization.md

```diff
@@ -0,0 +1,277 @@
+# ACK Handling Server Side Defaults
+
+## Problem
+
+During reconciliation, ACK service controller consumes the custom resource spec (i.e. desired state) from “etcd” and
+then creates/updates the AWS resource to match that desired state. ACK service controller uses the custom resource spec
+fields to construct Create/Update AWS API request.
+
+However, some AWS API’s behavior does not completely rely on the fields present in Create/Update AWS API request alone.
+In the backend, these AWS APIs either add new fields with default values or update the value of field present in request
+with newly calculated value which represents the latest state of the resource in AWS. AWS APIs return these default or
+updated values with Create/Update API output and Read/List/Describe API output after resource creation.
+
+There are two main problems arise due to this behavior of AWS APIs,
+
+a) When an AWS API does not strictly use the property value as is in the backend, but updates it, the desired spec of
+the K8s resource in the “etcd” does not correctly represent the actual state of resource in AWS. This is a problem because
+for K8s user, the spec of custom resource is the source of truth for state of K8s resource.
+
+b) When a default field is populated during AWS resource creation, then during reconciliation, there will always be delta
+present between desired and latest K8s representation of the custom resource and an Update AWS API call be made during
+reconciliation. This update call may or may not have any side-effects on AWS resource and currently service teams either
+place custom code to avoid this OR let the calls happen if it does not alter state of the resource.
+
+> Problem (b) can be viewed as a variation of Problem (a), where server changes the property value from “null” to something.
+This is worth noting because solving Problem (a) in a generic and consistent manner will provide the solution for
+Problem (b) as well.
+
+
+## Solutions
+
+To compare the solutions for above problem, we will use the following facets:
+
+1. Consistency with K8s conventions
+2. Consistency across ACK service controllers
+3. GitOps compatibility
+4. User experience for custom resource lifecycle 
+
+✅ → doable with no/some efforts
+
+❌ → not achievable
+
+### Solution 1
+
+In this solution, the status fields will include all the fields which are CreateOutput shape members. Current status 
+fields are the CreateOutput shape members which are not present in CreateInput shape (spec fields).
+
+During reconciliation, the Read(One|Many),Update request inputs will be created from desired state but giving precedence
+to status fields over spec in case of conflict.
```

Comment:
This case seems to imply that there could be fields in `status` with the same name as ones in `spec`? The ones in `status` are then given precedence, which would mean we never use the values in `spec` again. Therefore the `spec` fields are only referenced for the initial `create` and never again? 

Or are you implying here that the `status` fields are simply used by the `diff` and for UX purposes, and that they're not being used as the source of truth for update operations?

Reply:
They were supposed to be used as source of truth in this solution. But thankfully we are not implementing this solution.